server/pulp/server/async/worker_watcher.py

```diff
@@ -72,14 +71,12 @@ def handle_worker_heartbeat(event):
     :type event: dict
     """
     event_info = _parse_and_log_event(event)
-    worker = Worker.objects(name=event_info['worker_name']).first()
-
-    if not worker:
+    existing_worker, created = Worker.objects.get_or_create(name=event_info['worker_name'])
+    if created:
         msg = _("New worker '%(worker_name)s' discovered") % event_info
         _logger.info(msg)
-
-    Worker.objects(name=event_info['worker_name']).\
-        update_one(set__last_heartbeat=event_info['local_received'], upsert=True)
+    else:
+        existing_worker.save()
```

Comment:
Is this done to increment a timestamp as a side effect? I think it's worth an in-line comment saying so. Otherwise this appears unnecessary.